blockchain/src/block.rs

```diff
@@ -167,15 +168,13 @@ impl KeyBlock {
     pub fn new(
         base: BaseBlockHeader,
         leader: SecurePublicKey,
-        witnesses: &[SecurePublicKey],
+        witnesses: BTreeSet<SecurePublicKey>,
     ) -> Self {
-        let mut witnesses = witnesses.to_vec();
-
-        // Witnesses list must be sorted.
-        witnesses.sort();
+        // The set of witnesses must be non-empty.
+        assert!(!witnesses.is_empty());
 
         // Leader must present in witnesses array.
-        //assert_eq!(witnesses.binary_search(leader), Ok((_, _)));
+        assert!(witnesses.contains(&leader));
```

Comment:
can we put comments to assert message ?
```suggestion
        assert!(witnesses.contains(&leader), "Leader must present in witnesses array.");
```

Reply:
Thanks for suggestion! I didn't know that assert!() has second argument. I will update this commit.